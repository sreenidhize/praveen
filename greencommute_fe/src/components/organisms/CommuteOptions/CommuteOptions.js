import React, { useState } from "react";
import { makeStyles, Typography, Grid } from "@material-ui/core";
import PropTypes from "prop-types";
import CabList from "../../molecules/CabList/CabList";
import TransportMode from "../../molecules/TransportMode/TransportMode";
import CommuteRouteIcon from "../../molecules/CommuteRouteIcons/CommuteRouteIcons";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "350px",
  },
  optionsLabel: {
    fontSize: "16px",
    color: theme.typography.body1.color,
  },
  routeLabel: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
    color: theme.typography.body1.color,
  },
  mapText: {
    paddingTop: theme.spacing(4),
    color:theme.palette.primary.main,
    textAlign: "left",
    cursor: "pointer"
  },
  details: {
    paddingTop: theme.spacing(2),
    fontSize: theme.typography.subtitle2.fontSize,
    color: theme.typography.body1.color,
  },

}));

const defaultCommutes = [
  { 
    type: "metro" 
  },
  { 
    type: "bus" 
  },
  { 
    type: "car" 
  },
  { 
    type: "bike" 
  },
];

const selectCommute = (type) => {
  return defaultCommutes.map((item) => {
    item.type === type ? (item.selected = true) : (item.selected = false);
    return item;
  });
};

const CommuteOptions = ({ commuteTypes, selectedCommute }) => {
  const [commute, setCommute] = useState(selectedCommute);
  const icons = selectCommute(commute);

  const classes = useStyles();

  const changeCommute = (type) => {
    setCommute(type);
  };

  return (
    <Grid container  direction="column"  className={classes.root} data-testid="commute-options" >
      <Grid item container justify="space-between" alignItems='center'>
        <Typography variant={"h4"} className={classes.optionsLabel}>
          Your Options
        </Typography>
        <CommuteRouteIcon  icons={icons} onClick={changeCommute}/>
      </Grid>   
      
      {commute !== "car" ? (
        <div>
          <Grid item container justify="space-between">
            <Typography variant="h6" className={classes.routeLabel}>
              Maredpally - Hitech City
            </Typography>
            <Typography variant={"subtitle2"} className={classes.details}>
              $50 &bull; 56 mins
            </Typography>
          </Grid>
          <Grid item>
            {commuteTypes.map((item) => (
              <TransportMode
                description={item.description}
                type={item.type}
                fare={item.fare}
                isLate={item.isLate}
                status={item.status}
                time={item.time}
              />
            ))}
          </Grid>
          <Grid item>
            <Typography  variant="h5" className={classes.mapText} >
            View in google maps
            </Typography> 
          </Grid>
        </div>
      ) : (
        <Grid item>
          <CabList/>
        </Grid>
      )}
    </Grid>
  );
};

CommuteOptions.propTypes = {
  commuteTypes: PropTypes.array.isRequired,
};

export default CommuteOptions;
