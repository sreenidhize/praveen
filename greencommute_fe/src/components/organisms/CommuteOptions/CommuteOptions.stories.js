import React from "react";
import CommuteOptions from "./CommuteOptions";

export default {
  title: "Organisms/CommuteOptions",
  component: CommuteOptions,
};

const Template = (args) => <CommuteOptions {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  commuteTypes: [
    {
      type: "bus",
      fare: "15$",
      time: "20 mins",
      description: "Catch a bus 38X at 3.42 PM to Secunderabad bus stand.",
      status: "2 mins",
      isLate: true,
    },
    {
      type: "walk",
      time: "5 mins",
      description: "Head north on St. Johns road.",
    },
    {
      type: "metro",
      fare: "56$",
      time: "38 mins",
      description: "Catch a blue line metro towards Raidurg.",
      isLate: false,
    },
  ],
};
