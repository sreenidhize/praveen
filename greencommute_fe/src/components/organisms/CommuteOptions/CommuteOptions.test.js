import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CommuteOptions from "./CommuteOptions";

describe("Commute options", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <CommuteOptions commuteTypes={[]}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test commute options render", () => {
        const { getByTestId } = render(<CommuteOptions  commuteTypes={[]}/>);
        expect(getByTestId("commute-options")).toBeInTheDocument();
    });
});