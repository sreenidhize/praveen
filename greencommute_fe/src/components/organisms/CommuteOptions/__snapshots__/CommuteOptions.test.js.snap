// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commute options Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column"
    data-testid="commute-options"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <h4
        class="MuiTypography-root makeStyles-optionsLabel-2 MuiTypography-h4"
      >
        Your Options
      </h4>
      <div
        data-testid="commuteRoutes"
      >
        <div
          class="makeStyles-display-7"
        >
          <div>
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-12 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-8 makeStyles-iconUnselected-9"
                data-testid="metro"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="8.5"
                  cy="14.5"
                  r="1.5"
                />
                <circle
                  cx="15.5"
                  cy="14.5"
                  r="1.5"
                />
                <path
                  d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-12 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-8 makeStyles-iconUnselected-9"
                data-testid="bus"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-12 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-8 makeStyles-iconUnselected-9"
                data-testid="car"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.92 5.01C18.72 4.42 18.16 4 17.5 4h-11c-.66 0-1.21.42-1.42 1.01L3 11v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.85 6h10.29l1.04 3H5.81l1.04-3zM19 16H5v-4.66l.12-.34h13.77l.11.34V16z"
                />
                <circle
                  cx="7.5"
                  cy="13.5"
                  r="1.5"
                />
                <circle
                  cx="16.5"
                  cy="13.5"
                  r="1.5"
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-12 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-8 makeStyles-iconUnselected-9"
                data-testid="bike"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5zm5.8-10l2.4-2.4.8.8c1.3 1.3 3 2.1 5.1 2.1V9c-1.5 0-2.7-.6-3.6-1.5l-1.9-1.9c-.5-.4-1-.6-1.6-.6s-1.1.2-1.4.6L7.8 8.4c-.4.4-.6.9-.6 1.4 0 .6.2 1.1.6 1.4L11 14v5h2v-6.2l-2.2-2.3zM19 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
      >
        <h6
          class="MuiTypography-root makeStyles-routeLabel-3 MuiTypography-h6"
        >
          Maredpally - Hitech City
        </h6>
        <h6
          class="MuiTypography-root makeStyles-details-5 MuiTypography-subtitle2"
        >
          $50 â€¢ 56 mins
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      />
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root makeStyles-mapText-4 MuiTypography-h5"
        >
          View in google maps
        </h5>
      </div>
    </div>
  </div>
</div>
`;
